flash/data/process.py

```diff
@@ -101,6 +101,151 @@ class PreprocessState:
 
 
 class Preprocess(Properties, torch.nn.Module):
+    """
+    The :class:`~flash.data.process.Preprocess` is used to encapsulate
+    all the processing data loading logic up to the model.
+
+    It is particularly relevant when you want to provide ready-to-production implementation which works
+    with 4 different stages: ``training``, ``validation``, ``test` and inference(``predicting``).
+
+    The :class:`~flash.data.process.Preprocess` supports the following hooks:
+
+        - ``load_data``: Function to receiving some metadata to generate a Mapping from.
+            Example::
+
+                * Input: Receive a folder path:
+
+                * Action: Walk the folder path and find image paths and their associated labels.
+
+                * Output: Return a list of image paths and their associated labels.
+
+        - ``load_sample``: Function to load a sample from metadata sample.
```

Comment:
It's not clear to me what "metadata sample" is

Reply:
I have to say both these methods feel almost too arbitrary? I understand we want to keep this flexible, but we need to have some way of explaining what is expected.

Maybe "Given a pointer to a sample of data, returns the data in a X format? "